src/android/com/adobe/phonegap/push/PushPlugin.java

```diff
@@ -65,21 +65,7 @@ public void run() {
 
                         Log.v(LOG_TAG, "execute: senderID=" + senderID);
 
-                        String savedSenderID = sharedPref.getString(SENDER_ID, "");
-                        String savedRegID = sharedPref.getString(REGISTRATION_ID, "");
-
-                        // first time run get new token
-                        if ("".equals(savedRegID)) {
-                            token = InstanceID.getInstance(getApplicationContext()).getToken(senderID, GCM);
-                        }
-                        // new sender ID, re-register
-                        else if (!savedSenderID.equals(senderID)) {
-                            token = InstanceID.getInstance(getApplicationContext()).getToken(senderID, GCM);
-                        }
-                        // use the saved one
-                        else {
-                            token = sharedPref.getString(REGISTRATION_ID, "");
-                        }
+                        token = FirebaseInstanceId.getInstance().getToken();
```

Comment:
The senderID is part of the app id, so Firebase probably only loads that
value once.

Summers Pittman
>>Phone:404 941 4698
>>Java is my crack.


On Fri, Jun 10, 2016 at 7:30 PM, jcesarmobile <notifications@github.com>
wrote:

> In src/android/com/adobe/phonegap/push/PushPlugin.java
> <https://github.com/phonegap/phonegap-plugin-push/pull/975#discussion_r66694284>
> :
>
> > -                        String savedSenderID = sharedPref.getString(SENDER_ID, "");
> > -                        String savedRegID = sharedPref.getString(REGISTRATION_ID, "");
> > -
> > -                        // first time run get new token
> > -                        if ("".equals(savedRegID)) {
> > -                            token = InstanceID.getInstance(getApplicationContext()).getToken(senderID, GCM);
> > -                        }
> > -                        // new sender ID, re-register
> > -                        else if (!savedSenderID.equals(senderID)) {
> > -                            token = InstanceID.getInstance(getApplicationContext()).getToken(senderID, GCM);
> > -                        }
> > -                        // use the saved one
> > -                        else {
> > -                            token = sharedPref.getString(REGISTRATION_ID, "");
> > -                        }
> > +                        token = FirebaseInstanceId.getInstance().getToken();
>
> On my sample project I have this string on res/values/strings.xml (created
> by the plugin) <string name="google_app_id">myAppId3333</string> and I'm
> able to receive push notifications sent from FCM console, but I have
> nothing for gcm_defaultSenderId anywhere
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/phonegap/phonegap-plugin-push/pull/975/files/216ab837fd239c52b4a477f42ecff26a835aca74#r66694284>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ACTU-EPL8vOnB2vvra9nBLm8knk2r04wks5qKfOlgaJpZM4Iyd07>
> .
>